ml4h/applications/ingest/ingest_mri.py

```diff
@@ -0,0 +1,196 @@
+sample_manifest['Imaging Frequency'] = sample_manifest['Imaging Frequency'].astype(np.float32)
+sample_manifest['Instance Creation Date'] = pd.to_datetime(sample_manifest['Instance Creation Date'])
+sample_manifest['Instance Number'] = sample_manifest['Instance Number'].astype(np.int8)
+sample_manifest['Largest Image Pixel Value'] = sample_manifest['Largest Image Pixel Value'].astype(np.int16)
+sample_manifest['Magnetic Field Strength'] = sample_manifest['Magnetic Field Strength'].astype(np.float32)
+sample_manifest['Number of Averages'] = sample_manifest['Number of Averages'].astype(np.float32)
+sample_manifest['Number of Phase Encoding Steps'] = sample_manifest['Number of Phase Encoding Steps'].astype(np.int16)
+sample_manifest["Patient's Birth Date"] = pd.to_datetime(sample_manifest["Patient's Birth Date"])
+sample_manifest["Patient's Size"] = sample_manifest["Patient's Size"].astype(np.float32)
+sample_manifest["Patient's Weight"] = sample_manifest["Patient's Weight"].astype(np.float32)
+sample_manifest['Percent Phase Field of View'] = sample_manifest['Percent Phase Field of View'].astype(np.float32)
+sample_manifest['Percent Sampling'] = sample_manifest['Percent Sampling'].astype(np.float32)
+sample_manifest['Performed Procedure Step Start Date'] = pd.to_datetime(sample_manifest['Performed Procedure Step Start Date'])
+sample_manifest['Pixel Bandwidth'] = sample_manifest['Pixel Bandwidth'].astype(np.float32)
+sample_manifest['Pixel Representation'] = sample_manifest['Pixel Representation'].astype(np.float32)
+sample_manifest['Repetition Time'] = sample_manifest['Repetition Time'].astype(np.float32)
+sample_manifest['Rows'] = sample_manifest['Rows'].astype(np.int16)
+sample_manifest['SAR'] = sample_manifest['SAR'].astype(np.float32)
+sample_manifest['Samples per Pixel'] = sample_manifest['Samples per Pixel'].astype(np.float32)
+sample_manifest['Series Date'] = pd.to_datetime(sample_manifest['Series Date'])
+sample_manifest['Series Number'] = sample_manifest['Series Number'].astype(np.int16)
+sample_manifest['Slice Location'] = sample_manifest['Slice Location'].astype(np.float32)
+sample_manifest['Slice Thickness'] = sample_manifest['Slice Thickness'].astype(np.float32)
+sample_manifest['Smallest Image Pixel Value'] = sample_manifest['Smallest Image Pixel Value'].astype(np.int16)
+sample_manifest['Study Date'] = pd.to_datetime(sample_manifest['Study Date'])
+sample_manifest['Study ID'] = sample_manifest['Study ID'].astype(np.int32)
+sample_manifest['Slice Thickness'] = sample_manifest['Slice Thickness'].astype(np.float32)
+sample_manifest['Window Width'] = sample_manifest['Window Width'].astype(np.float32)
+sample_manifest['dB/dt'] = sample_manifest['dB/dt'].astype(np.float32)
+
+# Reformat column names to be all lower case and replace spaces with underscores
+sample_manifest.columns = [s.lower().replace(' ','_') for s in sample_manifest.columns.values]
+
+
+# Store meta data
+fp.write('body_mri_manifest',
```

Comment:
If we want to parallelize will appending be a problem?

Reply:
A 2-pass would be appropriate in the embarrassingly parallel case. We could modify this to saving individual Parquet files in `simple` mode --- when all these are finished we could merge them as above in `hive` mode.

Reply:
It's now in simple mode. A second pass will be required to reduce the map.